src/Foundation/NSUrl.cs

```diff
@@ -142,6 +142,19 @@ public bool SetResource (NSString nsUrlResourceKey, NSObject value)
 			return SetResourceValue (value, nsUrlResourceKey, out error);
 		}
 
+		public int Port
+		{
+			get
+			{
+				var number = this.PortNumber;
+
+				if (number == null)
+					return -1;
+
+				return (int)number;
```

Comment:
I believe you can re-write the code as:
```csharp
    get 
    {
        return (this.PortNumber == null) ? -1: (int) this.PortNumber;
    }
```
If getting the PortNumber is an expensive operation you can use a temp var and do the same unary operation.